Packages/com.unity.inputsystem/Tests/InputSystem/CoreTests_Actions.cs

```diff
@@ -1108,13 +1108,74 @@ public void Actions_WithMultipleBoundControls_DriveInteractionsFromControlWithGr
         }
     }
 
-    // There can be situations where two different controls are driven from the state. Most prominently, this is
+    [Test]
+    [Category("Actions")]
+    public void Actions_WithMultipleBoundControls_CanHandleInteractionsThatTriggerOnlyOnButtonRelease()
+    {
+        var keyboard = InputSystem.AddDevice<Keyboard>();
+        var gamepad = InputSystem.AddDevice<Gamepad>();
+
+        // Action with PressInteraction set to ReleaseOnly.
+        // In this configuration, PressInteraction will not do anything on button down. Make sure that
+        // the disambiguation code does not get confused by this but still correctly processes the button
+        // up state change (which is a state change *back* to default state, i.e. has zero magnitude).
+        var action = new InputAction(interactions: "Press(behavior=1)");
+
+        action.AddBinding("<Keyboard>/a");
+        action.AddBinding("<Gamepad>/buttonNorth");
+
+        action.Enable();
+
+        using (var trace = new InputActionTrace())
+        {
+            trace.SubscribeToAll();
+
+            Press(gamepad.buttonNorth);
+
+            Assert.That(trace, Is.Empty);
+
+            Release(gamepad.buttonNorth);
+
+            var actions = trace.ToArray();
+            Assert.That(actions, Has.Length.EqualTo(1));
+            Assert.That(actions[0].phase, Is.EqualTo(InputActionPhase.Performed));
+            Assert.That(actions[0].control, Is.SameAs(gamepad.buttonNorth));
+
+            trace.Clear();
+
+            Press(gamepad.buttonNorth);
+
+            Assert.That(trace, Is.Empty);
+
+            Release(gamepad.buttonNorth);
+
+            actions = trace.ToArray();
+            Assert.That(actions, Has.Length.EqualTo(1));
+            Assert.That(actions[0].phase, Is.EqualTo(InputActionPhase.Performed));
+            Assert.That(actions[0].control, Is.SameAs(gamepad.buttonNorth));
+
+            trace.Clear();
+
+            Press(keyboard.aKey);
+
+            Assert.That(trace, Is.Empty);
+
+            Release(keyboard.aKey);
+
+            actions = trace.ToArray();
+            Assert.That(actions, Has.Length.EqualTo(1));
+            Assert.That(actions[0].phase, Is.EqualTo(InputActionPhase.Performed));
+            Assert.That(actions[0].control, Is.SameAs(keyboard.aKey));
```

Comment:
What happens (or what should happen), if I press buttonNorth, then a, and then release buttonNorth, and then a? or  press buttonNorth, then a, and then release a, and then buttonNorth? On which releases will it trigger? Can we make the test cover that behavior?

Reply:
>What happens (or what should happen), if I press buttonNorth, then a, and then release buttonNorth, and then a?

1. Reacts to press of `buttonNorth`.
2. Does not react to press of `a` (same level of actuation as `buttonNorth` so does not result in change on the action).
3. It switches from `buttonNorth` to `a` when `buttonNorth` is released (sticks to control with greatest level of actuation).
4. It reacts to the release of `a`.

>or press buttonNorth, then a, and then release a, and then buttonNorth?

The in-between press and release of `a` has no effect. It has the same level of actuation as `buttonNorth` and thus will not get to drive the action as `buttonNorth` is already driving it. If it had *greater* actuation than `buttonNorth`, it would take over and temporarily drive the action until it is released (as then `buttonNorth` has the greater level of actuation).

Sounds complicated this way. Put simply, when there are multiple controls driving an action, the code attempts to always stick to the control with the greatest level of actuation. So as long as any control is still actuated, the action keeps going.

>Can we make the test cover that behavior?

Might be worth having a dedicated test for this. Not sure. ATM the test that covers the disambiguation described above is tested by `Actions_WithMultipleBoundControls_DriveInteractionsFromControlWithGreatestActuation`.

Reply:
Ok, fleshed out the test and covered the "two buttons being pressed and then released one after the other" scenario.